### PR TITLE
Added MIDI tab & keycodes

### PR DESCRIPTION
**Pull request by jkutianski**
To use this you've to add MIDI_ENABLE = yes to rules.mk, #define MIDI_BASIC or #define MIDI_ADVANCED to config.h & "vial": { "midi": "basic" }, or "vial": { "midi": "advanced" }, to vial.json

### CODE REVIEW

File: src/main/python/keyboard_comm.py [hunk 1]
```diff
@@ -306,6 +307,8 @@ def reload_layout(self, sideload_json=None):
 
         self.custom_keycodes = payload.get("customKeycodes", None)
 
+        self.midi = payload.get("midi", None)
```
Comment (xyzz) [inline src/main/python/keyboard_comm.py:310]: Put this under "vial" as it's a vial extension (see where `self.vibl` is currently assigned)

File: src/main/python/keycodes.py [hunk 2]
```diff
@@ -611,6 +769,16 @@ def create_custom_user_keycodes(custom_keycodes):
             )
         )
 
+def create_midi_keycodes(midiSettingLevel):
+    KEYCODES_MIDI.clear()
+
+    if midiSettingLevel == "basic" or midiSettingLevel == "advanced":
+        for x, kc in enumerate(KEYCODES_MIDI_BASIC):
```
Comment (xyzz) [inline src/main/python/keycodes.py:776]: `KEYCODES_MIDI.extend(KEYCODES_MIDI_BASIC)`

File: src/main/python/keycodes.py [hunk 3]
```diff
@@ -611,6 +769,16 @@ def create_custom_user_keycodes(custom_keycodes):
...
+            KEYCODES_MIDI.append(kc)
+
+    if midiSettingLevel == "advanced":
+        for x, kc in enumerate(KEYCODES_MIDI_ADVANCED):
```
Comment (xyzz) [inline src/main/python/keycodes.py:780]: same as above

**Review by xyzz — CHANGES_REQUESTED** [review]

File: src/main/python/keyboard_comm.py (hunk 1 above)
Comment (jkutianski) [reply src/main/python/keyboard_comm.py]: DONE!

File: src/main/python/keycodes.py (hunk 2 above)
Comment (jkutianski) [reply src/main/python/keycodes.py]: DONE!

File: src/main/python/keycodes.py (hunk 3 above)
Comment (jkutianski) [reply src/main/python/keycodes.py]: DONE!

**Pull request merged**